File: lib/entities/sadad_transaction.dart
import 'package:sadad_flutter_sdk/entities/sadad_order_item.dart';

class SadadTransaction {
  final String mobileNumber;
  final String customerId;
  final String clientToken;
  final String orderId;
  final String transactionAmount;
  final String email;
  final List<SadadOrderItem> orderItems;

  const SadadTransaction({
    required this.mobileNumber,
    required this.email,
    required this.customerId,
    required this.clientToken,
    required this.transactionAmount,
    required this.orderItems,
    required this.orderId,
  }) : assert(orderItems.length > 0);

  Map<String, dynamic> toJson() => {
        "mobileNumber": mobileNumber,
        "customerId": customerId,
        "clientToken": clientToken,
        "orderId": orderId,
        "transactionAmount": transactionAmount,
        "orderItems": orderItems.map((e) => e.toJson()).toList(),
        "email": email
      };
}

File: android/src/main/kotlin/com/example/sadad_flutter_sdk/entities/SadadTransaction.kt
package com.example.sadad_flutter_sdk.entities

import org.json.JSONObject


data class SadadTransaction(
    val mobileNumber: String,
    val customerId: String,
    val clientToken: String,
    val orderId: String,
    val transactionAmount: String,
    val orderItems: List<SadadOrdrItem>
) {
    constructor(
        json: JSONObject,
        orderItems: List<SadadOrdrItem>
    ) : this(
        json["mobileNumber"] as String,
        json["customerId"] as String,
        json["clientToken"] as String,
        json["orderId"] as String,
        json["transactionAmount"] as String,
        orderItems
    )
}